bindings/rabbitmq/rabbitmq.go

```diff
@@ -84,34 +96,78 @@ func (r *RabbitMQ) Init(_ context.Context, metadata bindings.Metadata) error {
 		return err
 	}
 
-	conn, err := amqp.Dial(r.metadata.Host)
+	err = r.connect()
 	if err != nil {
 		return err
 	}
+	r.notifyRabbitChannelClose = make(chan *amqp.Error, 1)
+	r.channel.NotifyClose(r.notifyRabbitChannelClose)
+	go r.reconnectWhenNecessary()
+	return nil
+}
 
-	ch, err := conn.Channel()
-	if err != nil {
-		return err
+func (r *RabbitMQ) reconnectWhenNecessary() {
+	for {
+		select {
+		case <-r.closeCh:
+			return
+		case _ = <-r.notifyRabbitChannelClose:
+			// this is called when the server restart or the channel is closed by server.
+			if r.connection != nil && !r.connection.IsClosed() {
```

Comment:
This probably needs to be wrapped in a lock or it can cause race conditions

Reply:
Done.